python_training/mimic.py

```diff
@@ -0,0 +1,122 @@
+
+class MyList:
+    """
+    Creates a series of objects that mimic a pythonic list.
+    Each object holds a value, and a pointer to the next item of the list.
+    The user accesss the list through the first item.
+    The first initiation of the class (by the user) creates an empty list, and items are added by calling the append
+    method.
+    """
+    def __init__(self, index=0, value=None):
+        """
+        Creates a list member that holds a specific value of the list, and points to the next value.
+        At first self.next_value is empty, until the append method is called.
+        :param index: The index of the item in the series
+        :param value: The value in the series that the specific MyList object represents.
+        """
+        self.value = value
+        self.next_value = None
+        self.index = index
+
+    def append(self, appended_value):
+        """
+        Appends a value to the series. If empty, the value of the current object would be set.
+        Otherwise, the method would be called upon the next object.
+        :param appended_value: The value that would be added to the series.
+        """
+        if self.value is None:
+            self.value = appended_value
+        elif self.next_value is None:
+            self.next_value = (MyList(self.index + 1, appended_value))
+        else:
+            self.next_value.append(appended_value)
+
+    def __len__(self)-> int:
+        """
+        Calculates the number of objects in the series.
+        :return: The number of objects in the series.
+        """
+        return self._get_last_index() + 1
+
+    def _get_last_index(self)->int:
+        """
+        Finds out what is the index of the last object in the series.
+        :return: The index of the last object.
+        """
+        if self.next_value is None:
```

Comment:
Nice one-liner:
`last_index = self.index if self.next_value is None else self.next_value._get_last_index()`